dbms/src/Storages/Page/V3/PageDirectoryFactory.cpp

```diff
@@ -162,7 +162,7 @@ void PageDirectoryFactory::applyRecord(
             if (holder)
             {
                 *holder = r.page_id;
-                dir->external_ids.emplace_back(std::weak_ptr<PageIdV3Internal>(holder));
+                dir->external_ids_by_ns.addExternalId(holder);
```

Comment:
Why use `addExternalId` here but `addExternalIdUnlock` above?

Reply:
Should also use `addExternalIdUnlock`. Fixed